services/wfm/src/main/java/org/openkilda/wfm/share/service/SpeakerFlowSegmentRequestBuilder.java

```diff
@@ -103,22 +103,21 @@ public SpeakerFlowSegmentRequestBuilder(FlowResourcesManager resourcesManager) {
 
         List<FlowSegmentRequestFactory> requests = new ArrayList<>();
         requests.addAll(makePathRequests(
-                path, context, encapsulation, doIngress, doTransit, doEgress));
+                flow, path, context, encapsulation, doIngress, doTransit, doEgress));
         if (oppositePath != null) {
-            if (!flow.isOneSwitchFlow() && oppositePath != null) {
+            if (!flow.isOneSwitchFlow()) {
                 encapsulation = getEncapsulation(
                         flow.getEncapsulationType(), oppositePath.getPathId(), path.getPathId());
             }
             requests.addAll(makePathRequests(
-                    oppositePath, context, encapsulation, doIngress, doTransit, doEgress));
+                    flow, oppositePath, context, encapsulation, doIngress, doTransit, doEgress));
         }
         return requests;
     }
 
     private List<FlowSegmentRequestFactory> makePathRequests(
-            @NonNull FlowPath path, CommandContext context, FlowTransitEncapsulation encapsulation,
+            Flow flow, FlowPath path, CommandContext context, FlowTransitEncapsulation encapsulation,
```

Comment:
Why do we allow to build flow segment requests for "null" path?

Reply:
We don't. This is a private method, I have doubts regarding the usefulness of `@NonNull` annotations for private methods.